### ISSUE TITLE
Main Request execution has been completed before the Pre Request

### ISSUE THREAD

**Issue by gopival** [usebruno/bruno#3554]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

1. I need to execute a post request to generate with a specific payload to generate 2FA code in the form of Pre Request
2. Store the 2FA as a environment variable
3. Main request has to use this 2FA code from the environment to generate the access token.

In my case both Pre Request and Main request  is being executed parallely and not sequentially as given above. Due to this, Main Request is picking the 2FA from the previous run of the Pre Request instead the current run. Bottom line is Main Request execution is completed even before the Pre Request writes the 2FA to the environment. Could you please help me with this issue?

Pre Request:
const axios = require('axios');
var client_id = bru.getEnvVar("user_client_id")
var user = bru.getEnvVar("user_email1")
var password = bru.getEnvVar("user_password")
var base_url = bru.getEnvVar("base_url")
var url = base_url+"/v1/tokens"
var user_device_id = bru.getEnvVar("user_device_id1")
var device_id, code, tfa;
let isReady;

var headers = {
    "Content-Type": "application/json"
};

var data = JSON.stringify({
  "grant_type":"password",
  "client_id": client_id,
  "username": user,
  "password": password
});

async function sendRequest() {
  // Trigger the POST request using Axios
  await axios.post(url, data, { headers: headers })
    .then(response => {
        // Successfully received the response
        console.log('POST request response', response);
    })
    .catch(async function (error) {
        // Handle any errors in the POST request
      if (error.response.status === 403) {
        console.log('POST request is successful');
        console.info('RESP:', error.response.data);
        device_id = error.response.data.devices[0].id;
        code = error.response.data.message.match(/(\d+)/);
        console.info("Code: ", code[0])
        await bru.setEnvVar("user_device_id1", device_id, {persistent: true});
        await bru.setEnvVar("2fa1", code[0], {persistent: true});
        await console.info("2fa1", bru.getEnvVar("2fa1"))
        isReady = true;
      } else {
          isReady = false;
          console.log('POST request is unsuccessful ', error);
      }
    });
}

sendRequest().then(async function() {
  if (isReady) {
    await console.log("Pre-request finished, proceeding with the main request...");
    //bru.setNextRequest('TOKENS-USER-1');
  } else {
        await console.log("Pre-request failed, halting request execution.");
    }
});
 

### Screenshots/Live demo link

<img width="1170" alt="Screenshot 2024-11-27 at 12 35 16" src="https://github.com/user-attachments/assets/fa2e6b16-46b3-488f-8e6e-01842fd203ff">
<img width="919" alt="Screenshot 2024-11-27 at 12 43 47" src="https://github.com/user-attachments/assets/ae681f01-14c4-455e-b586-595bf1b40b3e">

**Comment by anusree-bruno** [comment]
@gopival
Thank you for reporting this issue.

I was able to reproduce the behavior you described. For me, the variables are not set or updated if used inside an async function. Is it the same for you?

Could you confirm if the variables are getting updated in the environment, but the request is not using the updated values for variables?

**Comment by gopival** [comment]
@anusree-bruno Yes, environment variables ('user_device_id1', '2fa1') are getting updated as expected. Actually I'm able to fetch the value of '2fa1' from the environment and print it in the console. I have the below statement in my script.
`
await console.info("2fa1", bru.getEnvVar("2fa1"))`

**Comment by sagarnishant1** [comment]
Yes i am also facing the same issue. i set accessToken inside prescript async axios call but that is not getting reflected in my actual request

**Comment by anusree-bruno** [comment]
Are you awaiting these async requests?

![Image](https://github.com/user-attachments/assets/266664d0-4084-459c-b6ac-4c0430880e97)

**Comment by anusree-bruno** [comment]
@gopival 
Can you await the sendRequest call and try?
sharing my working script

```
const axios = require('axios');

let isReady = false;
await bru.setNextRequest("echo numbers")
const headers = {
  'Content-Type': 'application/json',
  'x-api-key': 'reqres-free-v1',
  'check': 'again', 
  'collection-header': 'collection-header-value'  
};

const data = {
  email: 'eve.holt@reqres.in',
  password: 'cityslicka'
};

async function sendRequest() {
  try {
    const response = await axios.post('https://reqres.in/api/login', data, { headers });

    if (response.data && response.data.token) {
      await bru.setEnvVar('Token', bru.interpolate('{{$randomUUID}}'));
      console.log('✅ Token / 2FA Code:', response.data.token);
      isReady = true;
    } else {
      console.warn('⚠️ Login succeeded, but no token returned:', response.data);
    }
  } catch (err) {
    console.error('❌ Login failed:', err.response?.data || err.message);
  }
}

await sendRequest().then(async () => {
  if (isReady) {
    console.log('✅ Pre-request completed, proceeding to main request...');
    bru.setNextRequest('test-2'); 
  } else {
    console.log('❌ Pre-request failed, halting.');
  }
});

```

https://github.com/user-attachments/assets/33c4e720-d495-4ac0-940f-30f901678b11
